src/modules/zenoh/subscribers/uorb_subscriber.hpp

```diff
@@ -65,6 +65,8 @@ class uORB_Zenoh_Subscriber : public Zenoh_Subscriber
 	{
 		char data[_uorb_meta->o_size];
 
+		// TODO process rmw_zenoh attachment
+
 		const z_loaned_bytes_t *payload = z_sample_payload(sample);
```

Comment:
`payload` can't directly be used as a buffer, since it's a list of slices and the values at that address are nonsense. The best performance would be through checking if `z_bytes_get_contiguous_view` succeeds and then using that slice directly, or else using `z_bytes_reader_read` to read into a temporary array here.

Reply:
Thanks, I've mostly been focusing on PX4 - > ROS2, and I see Zenoh Api changed a bit as well. Will update this accordingly.